Sources/FoundationExtensions.swift

```diff
@@ -16,6 +16,45 @@ import struct Result.AnyError
 	import let CDispatch.NSEC_PER_SEC
 #endif
 
+extension FileHandle: ReactiveExtensionsProvider {}
+
+#if !os(Linux)
+extension Reactive where Base: FileHandle {
+	/// Returns a SignalProducer that reads the contents of this file in the
+	/// background.
+	///
+	/// - parameters:
+	///   - scheduler: A run loop scheduler to perform the asynchronous file read.
+	///
+	/// - returns: A SignalProducer of the file's data.
+	public func readToEndOfFile(on scheduler: RunLoopScheduler) -> SignalProducer<Data, NoError> {
```

Comment:
`NoError` doesn't seem correct here. The documentation is pretty light on details, but the `userInfo` can contain `NSFileHandleError`, "An `NSNumber` object containing an integer representing the UNIX-type error which occurred".

Reply:
Thanks for digging that up. I recall doing a search in the docs for error handling info at some point, but didn't find this!